BMM.Core/Implementations/DeepLinking/DeepLinkHandler.cs

```diff
@@ -178,7 +180,28 @@ private async Task PlayTrackById(TrackLinkParameters trackLinkParameters)
 
         public bool OpenFromInsideOfApp(Uri uri, string origin) => Open(uri, "internal link opened", origin);
 
-        public bool OpenFromOutsideOfApp(Uri uri) => Open(uri, "deep link opened");
+        public bool OpenFromOutsideOfApp(Uri uri)
+        {
+            if (_readyToHandleDeepLink)
+                return Open(uri, "deep link opened");
+
+            if (!_links.Any(l => l.PerformCanNavigateTo(uri, out _)))
+                return false;
+            
+            _pendingDeepLink = uri;
+            return true;
+        }
+
+        public void SetReadyToOpenDeepLinkAndHandlePending()
+        {
+            _readyToHandleDeepLink = true;
+            
+            if (_pendingDeepLink == null)
+                return;
+
+            OpenFromOutsideOfApp(_pendingDeepLink);
```

Comment:
we ignore the boolean return value.
should we log in case it's false?

Reply:
Before setting _pendingDeepLink in OpenFromOutsideOfApp we already handles the result, so it's no needed here.